ui-app/client/src/components/PageEditor/functions/PageOperations.ts

```diff
@@ -268,6 +268,143 @@ export class PageOperations {
 		}
 	}
 
+	private getCurrentCompEventKeys(componentKey: string, def: PageDefinition): Array<string> {
+		const cdDef = components.get(def.componentDefinition[componentKey].type ?? '');
+
+		if (!cdDef) {
+			return [];
+		}
+
+		const eventNames = cdDef?.properties
+			.filter(each => each.editor == ComponentPropertyEditor.EVENT_SELECTOR)
+			.map(each => each.name);
+
+		//clicked component event keys
+		const currentCompProperties: any = def.componentDefinition[componentKey].properties;
+
+		return eventNames
+			.filter(each => each in currentCompProperties)
+			.map(each => currentCompProperties[each].value);
+	}
+
+	private getEventOfAllChildren(componentKey: string, def: PageDefinition): string[] {
+		let eventKeys: Array<string> = [];
+
+		const que = new LinkedList<ComponentDefinition>();
+
+		que.add(def.componentDefinition[componentKey]);
+
+		while (que.size() > 0) {
+			const x = que.pop();
+			console.log(x);
+
+			if (!x.children) {
+				eventKeys = eventKeys.concat(this.getCurrentCompEventKeys(x.key, def));
+				continue;
+			}
+			Object.keys(x.children).forEach(k => {
+				const e = def.componentDefinition[k];
+				que.add(e);
+			});
+		}
+
+		console.log(eventKeys);
+
+		return eventKeys;
+	}
+
+	public deleteComponentWithEvents(componentKey: string | undefined) {
+		if (!componentKey || !this.defPath) return;
+
+		const pageDef: PageDefinition = getDataFromPath(
+			this.defPath,
+			this.locationHistory,
+			this.pageExtractor,
+		);
+		if (!pageDef) return;
+
+		if (pageDef.rootComponent === componentKey) {
+			// When a root component is delete we need to add a new grid so people can add components to it.
+			// It requires a confirmation if we can delete a grid that is root.
+			this.setIssue({
+				message:
+					'Deleting the root component will delete the entire screen. Do you want to delete?',
+				defaultOption: 'No',
+				options: ['Yes', 'No'],
+				callbackOnOption: {
+					Yes: () => {
+						if (!this.defPath) return;
+						let def: PageDefinition = getDataFromPath(
+							this.defPath,
+							this.locationHistory,
+							this.pageExtractor,
+						);
+						const key = this.genId();
+						def = {
+							...def,
+							rootComponent: key,
+							componentDefinition: {
+								[key]: { key, name: 'Page Grid', type: 'Grid' },
+							},
+						};
+						this.onSelectedComponentChanged('');
+						setData(this.defPath, def, this.pageExtractor.getPageName());
+					},
+				},
+			});
+		} else {
+			let def = {
+				...pageDef,
+				componentDefinition: { ...(pageDef.componentDefinition ?? {}) },
+			};
+
+			const clickedCompEventKeys: Array<string> = this.getEventOfAllChildren(
+				componentKey,
+				def,
+			);
+
```

Comment:
@Sumitabhbiswas can u please get the map of event keys instead of arrays itself?